### PR TITLE
pdclient: adjust get region by id

### PR DESCRIPTION
**Pull request by BusyJay**
Currently get_region_by_id will return an empty region if not found, which is very easy to lead to mistake. An explicit Option should be returned instead.

@huachaohuang @siddontang PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
Please add a description for your PR.

File: src/raftstore/store/worker/pd.rs [hunk 1]
```diff
@@ -258,6 +258,10 @@ impl<T: PdClient> Runner<T> {
                       pd_region);
                 PD_VALIDATE_PEER_COUNTER_VEC.with_label_values(&["peer valid"]).inc();
             }
+            Ok(None) => {
+                // split region has not yet report to pd.
+                // TODO: handle merge
```
Comment (hhkbp2) [inline src/raftstore/store/worker/pd.rs]: What is the meaning of this TODO?

Comment (BusyJay) [inline src/raftstore/store/worker/pd.rs]: Because None is returned, the region can be removed for either late split heartbeat or merged.

Comment (hhkbp2) [inline src/raftstore/store/worker/pd.rs]: The region merge would not cause region info disappear.
If region info is missing in PD, a retry would be appropriate for peer validation.

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
PTAL

**Pull request merged**